domain/src/main/java/org/oppia/android/domain/exploration/lightweightcheckpointing/ExplorationCheckpointController.kt

```diff
@@ -228,6 +235,31 @@ class ExplorationCheckpointController @Inject constructor(
     }
   }
 
+  fun isSavedCheckpointCompatibleWithExploration(
+    profileId: ProfileId,
+    explorationId: String
+  ): DataProvider<Boolean> {
+    return retrieveCacheStore(profileId)
+      .transformAsync(
+        CHECK_IS_EXPLORATION_CHECKPOINT_COMPATIBLE_WITH_EXPLORATION_DATA_PROVIDER_ID
+      ) { explorationCheckpointDatabase ->
+        val checkpoint = explorationCheckpointDatabase.explorationCheckpointMap[explorationId]
+        val chapterDataJsonObject = jsonAssetRetriever.loadJsonFromAsset("$explorationId.json")
```

Comment:
We should be using ExplorationRetriever here, instead, so that this can work correctly when loading from proto (we currently support loading topic data from both protos and json).

Reply:
done